x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/AsyncSearchTaskTests.java

```diff
@@ -155,56 +152,14 @@ public void onFailure(Exception e) {
         latch.await();
     }
 
-    public void testGetResponseFailureDuringReduction() throws InterruptedException {
-        AsyncSearchTask task = createAsyncSearchTask();
-        task.getSearchProgressActionListener().onListShards(Collections.emptyList(), Collections.emptyList(),
-            SearchResponse.Clusters.EMPTY, false);
-        InternalAggregations aggs = InternalAggregations.from(Collections.singletonList(new StringTerms("name", BucketOrder.key(true),
-            BucketOrder.key(true), 1, 1, Collections.emptyMap(), DocValueFormat.RAW, 1, false, 1, Collections.emptyList(), 0)));
-        //providing an empty named writeable registry will make the expansion fail, hence the delayed reduction will fail too
-        //causing an exception when executing getResponse as part of the completion listener callback
-        DelayableWriteable.Serialized<InternalAggregations> serializedAggs = DelayableWriteable.referencing(aggs)
-            .asSerialized(InternalAggregations::readFrom, new NamedWriteableRegistry(Collections.emptyList()));
-        task.getSearchProgressActionListener().onPartialReduce(Collections.emptyList(), new TotalHits(0, TotalHits.Relation.EQUAL_TO),
-            serializedAggs, 1);
-        AtomicReference<AsyncSearchResponse> response = new AtomicReference<>();
-        CountDownLatch latch = new CountDownLatch(1);
-        task.addCompletionListener(new ActionListener<>() {
-            @Override
-            public void onResponse(AsyncSearchResponse asyncSearchResponse) {
-                assertTrue(response.compareAndSet(null, asyncSearchResponse));
-                latch.countDown();
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                throw new AssertionError("onFailure should not be called");
-            }
-        }, TimeValue.timeValueMillis(10L));
-        assertTrue(latch.await(1, TimeUnit.SECONDS));
-        assertNotNull(response.get().getSearchResponse());
-        assertEquals(0, response.get().getSearchResponse().getTotalShards());
-        assertEquals(0, response.get().getSearchResponse().getSuccessfulShards());
-        assertEquals(0, response.get().getSearchResponse().getFailedShards());
-        assertThat(response.get().getFailure(), instanceOf(ElasticsearchException.class));
-        assertEquals("Async search: error while reducing partial results", response.get().getFailure().getMessage());
-        assertThat(response.get().getFailure().getCause(), instanceOf(IllegalArgumentException.class));
-        assertEquals("Unknown NamedWriteable category [" + InternalAggregation.class.getName() + "]",
-            response.get().getFailure().getCause().getMessage());
-    }
```

Comment:
is this test no longer relevant?

Reply:
it cannot work anymore since we don't need to serialize the aggs. I think it's ok since we have other tests that check that exception thrown during a partial/final reduce are handled correctly.

Reply:
I see, you mean the condition that the test had to trigger the failure, which was around serialization?

Reply:
yep